pkg/bbr/utils/init.go

```diff
@@ -0,0 +1,147 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Initializes PluginRegistry from environment variables (as set in the helm chart)
+
+package utils
+
+import (
+	"fmt"
+	"os"
+	"regexp"
+	"strings"
+
+	framework "sigs.k8s.io/gateway-api-inference-extension/pkg/bbr/framework"
+	bbrplugins "sigs.k8s.io/gateway-api-inference-extension/pkg/bbr/plugins"
+)
+
+func InitPlugins() (
+	framework.PluginRegistry,
+	framework.PluginsChain,
+	framework.PluginsChain,
+	error) {
+
+	//The environment variables defining plugins repertoire and plugin chains are set via Helm chart
+	registry := framework.NewPluginRegistry()
+	requestChain := framework.NewPluginsChain()
+	responseChain := framework.NewPluginsChain()
+
+	//Define a standardized regex patterns for plugin types
+	var pluginTypeRe = regexp.MustCompile(bbrplugins.PluginTypePattern)
+
+	//Define a standardized regex pattern for plugin names
+	var pluginNameRe = regexp.MustCompile(bbrplugins.PluginNamePattern)
+
+	//helper to validate plugin name
+	isValidPluginName := func(name string) bool {
+		if len(name) == 0 || len(name) > bbrplugins.MaxPluginNameLength {
+			return false
+		}
+		return pluginNameRe.MatchString(name)
+	}
+
+	//helper to validate plugin type name
+	isValidPluginType := func(name string) bool {
+		if len(name) == 0 || len(name) > bbrplugins.MaxPluginTypeLength {
+			return false
+		}
+		return pluginTypeRe.MatchString(name)
+	}
+
+	//helper to process plugins
+	processPlugin := func(pluginType string, chain framework.PluginsChain) error {
+
+		//create the plugin instance
+		plugin, err := registry.CreatePlugin(pluginType)
+		if err != nil {
+			return fmt.Errorf("failed to create an instance of %s %v", pluginType, err)
+		}
+
+		//register the plugin instance
+		err = registry.RegisterPlugin(plugin)
+		if err != nil {
+			return fmt.Errorf("failed to register an instance of %s %v", pluginType, err)
+		}
+
+		//Add plugin type name to the pluginsChain instance
+		err = chain.AddPlugin(pluginType, registry)
+		if err != nil {
+			return fmt.Errorf("failed to add plugin instance %s %v", pluginType, err)
+		}
+		return nil
+	}
+
+	// Helper to process plugin chains
+	processChain := func(envVar string, chain framework.PluginsChain) error {
+		envPluginsChain := os.Getenv(envVar)
+
+		if envPluginsChain == "" {
+			return nil // no plugins defined for this chain, but this is not an error
+		}
+
+		//Plugins are specified as, e.g., REQUEST_PLUGINS_CHAIN=MetaDataExtractor:simple-model-extractor, MyPluginType:my-plugin-name
```

Comment:
To follow up on @elevran's comment about envronment variables, why use a configuration along the lines of the EPP configuration?